### PR TITLE
Allow mixed precision TFT training

### PR DESCRIPTION
**Pull request by Marcrb2**
### Description

This PR modifies the attention mask in the TFT model from 1e-9 to float("inf") to allow Pytorch mixed precision training.

Related issues: 
#1325
#285

### CODE REVIEW

**Review by MrGG14 — APPROVED** [review]

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

From the API perspective, we need to address the problem that this changes default behaviour, and thus may break or change user code downstream.

What we need to do is to expose the constant in `masked_fill` as a parameter in `ScaledDotProductAttention`, up until `TemporalFusionTransformer`, that means also in `InterpretableMultiHeadAttention`.

The default must be left at 1e9 for now (to avoid breaking/changing other peoples' code), but this will allow you to set `float("inf")` by using the new parameter.

The docstrings of the modules and components should also get the description of the new parameter. Where classes have no docstring, it would be great if you could add one, but that is not blocking.

**Comment by fkiraly** [comment]
I would also appreciate feedback on which value is better as a default - even if we cannot change it right now, we could after a couple releases, giving the users a forewarning that the default will change.

**Comment by tr01-crypto** [comment]
is there an update to this issue? i would really need this feature..

**Comment by hima-erla** [comment]
Same, I really need this feature too. Mixed precision is an important aspect to be able to tune train heavy models like TFT. Can someone action this?

**Review by fkiraly — APPROVED** [review]
I think @Marcrb2 has disengaged, I've wrapped this up now.

Instead of changing the default, the `mask_bias` can now be manually set to -inf, which should be sufficiente for your use cases, @tr01-crypto, @hima-erla?
